src/firefly/js/ui/FitsDownloadDialog.jsx

```diff
@@ -147,19 +128,47 @@ function getInitState()  {
 }
 
 const FitsDownloadDialogForm= memo( ({isWs, popupId, groupKey}) => {
-    const [{pv,plot,hasOperation,threeC,colors}]= useState(getInitState);
-    useBindFieldGroupToStore(groupKey); // this will force a rerender when anything changes
+    const [{pv, plot, hasOperation, threeC, colors}] = useState(getInitState);
     const band = threeC ? getFieldVal(groupKey, 'threeBandColor', colors[0]) : Band.NO_BAND.key;
-    const totalChildren = (isWs ? 3 : 2) +  (hasOperation ? 1 : 0) + (threeC ? 1 : 0);// fileType + save as + (fileLocation)
-    const childH = (totalChildren*(20+mTOP));
+    const totalChildren = (isWs ? 3 : 2) + (hasOperation ? 1 : 0) + (threeC ? 1 : 0);// fileType + save as + (fileLocation)
+    const childH = (totalChildren * (20 + mTOP));
+
+    const [getFileType, setFileType] = useFieldGroupValue ('fileType', groupKey);
+    const [getFileName, setFileName] = useFieldGroupValue('fileName', groupKey);
+    const [getLocation, setLocation] = useFieldGroupValue('location', groupKey);
+
+    useEffect(() => {
+        const fileType = getFileType();
+        const fileName = getFileName();
+        const fileLocation = getLocation();
+        let fName = '';
+
+        // change the filename if a file is selected from the file picker
+        if (fileLocation === 'isWs' && isValidWSFolder(fileLocation, false).valid) {
+            fName = fileLocation.substring(fileLocation.lastIndexOf('/') + 1);
+        }
+        else { //isLocal: check for fileType change or fileName change. If fileType changes, replace file extension
+            if (!isThreeColor(plot)) {
+                //checking if filename is !empty string allows user to delete the input
+                if (fileName) fName = replaceExt(fileName, fileType);//replaceExt(fields.fileName.value, fType);
+            } else {
+                fName = matchPossibleDefaultNames(plot, fileName) ?
+                    makeFileName(plot, fileName, fileType) :
+                    replaceExt(fileName, fileType);
+            }
+        }
+        setFileName(fName);
+    }, [getFileType, getFileName, getLocation]);
```

Comment:
Personally, I find the behavior very awkward.
I bring up the dialog and wanted to change the name.  So, I select all, then begin to type... 'a', immediately it jumped to 'a.fits', at this point, I cannot backspace(delete) because my cursor is at the end and the field will not let me enter anything that does not end with '.fits'.  But, I don't know that.  I thought my keyboard is stuck.
But if I leave it blank, then I can process to save?  This is odd.  It let me save, but not a file name I chose.
Anyways, I understand this is existing behavior.  Therefore, this is just a comment and not a change request.